tests/testthat/test-check_lhs_has_only_zeros.R

```diff
@@ -0,0 +1,71 @@
+invalid_mat <- matrix(
+  c(
+    0, 0, 1, 2, 3,
+    0, 0, NA, NA, NA
+  ),
+  byrow = TRUE,
+  nrow = 2
+)
+valid_mat1 <- matrix(
+  c(
+    0, 0, 1, 2, 3,
+    0, 1, NA, NA, NA
+  ),
+  byrow = TRUE,
+  nrow = 2
+)
+valid_mat2 <- matrix(
+  c(
+    0, 0, 1, 2, 3,
+    1, 0, NA, NA, NA
+  ),
+  byrow = TRUE,
+  nrow = 2
+)
+
+invalid_mat2 <- matrix(
+  c(
+    0, 0, 4, 5, 6,
+    0, 0, 1, 2, 3,
+    0, 0, NA, NA, NA
+  ),
+  byrow = TRUE,
+  nrow = 3
+)
+valid_mat3 <- matrix(
+  c(
+    0, 0, 4, 5, 6,
+    0, 0, 1, 2, 3,
+    0, 0, 1, 2, 3
+  ),
+  byrow = TRUE,
+  nrow = 3
+)
+valid_mat4 <- matrix(
+  c(
+    4, 1, 4, 5, 6,
+    0, 0, 1, 2, 3,
+    0, 0, NA, NA, NA
+  ),
+  byrow = TRUE,
+  nrow = 3
+)
+edge_case_mat <- matrix(
+  c(
+    1, 2, 3, 4, 5,
+    NA, NA, NA, NA, NA
+  ),
+  byrow = TRUE,
+  nrow = 2
+)
+
+
+test_that(".check_lhs_for_zeros correctly handles matrix with 0s", {
```

Comment:
I'm not convinced we need to test this internal function at all and instead should have these all be tests of the validation step.